### ISSUE TITLE
[BUG] Setting override_main_response_version in opensearch.yml doesn't work after setting it via API without notice

### ISSUE THREAD

**Issue by DrissiReda** [opensearch-project/OpenSearch#4242]
**Describe the bug**
I'm trying to set `compatibility.override_main_response_version` setting inside my  `opensearch.yml` configuration file, but it is ignored.

However, I can set it successfully using the API:

```
PUT _cluster/settings
{
  "persistent": {
    "compatibility": {
      "override_main_response_version": true
    }
  }
}
```

I can see that I'm setting it properly from the last entry in the configuration outlined in this section of the official documentation: https://opensearch.org/docs/latest/opensearch/configuration/#configuration-file

**Host/Environment (please complete the following information):**
 - OS: Kubernetes
 - Version 1.3.4

**Comment by dblock** [comment]
I am guessing, but does this work?

```
PUT /_cluster/settings
{
  "persistent" : {
    "compatibility.override_main_response_version" : true
  }
}
```

**Comment by DrissiReda** [comment]
> I am guessing, but does this work? What's in the logs with either version?
> 
> ```
> PUT /_cluster/settings
> {
>   "persistent" : {
>     "compatibility.override_main_response_version" : true
>   }
> }
> ```

yes it works without any hiccups. And it persists reboots.

**Comment by dblock** [comment]
Oh I misread the issue. You're saying it saves, persists reboots, but doesn't work? How do you know it doesn't work?

**Comment by DrissiReda** [comment]
@dblock Yes I follow the documentation and it doesn’t work. I know that because a simple `GET /` returns version 1.3.4. But when I activate the settings through the API the returned version is 7.10.2.

**Comment by dblock** [comment]
@DrissiReda if you haven't yet, try on a generic clean installation/download of OpenSearch? I wonder whether setting it via API once causes this or something like that.  This should be debugged anyway.

**Comment by DrissiReda** [comment]
That’s what I did. Can’t find anything about it in logs when using the conf file.

**Comment by dblock** [comment]
It's ok that a cluster setting takes precedence. We need to 1) document it 2) log a warning that the setting in opensearch.yml will be ignored upon startup. WDYT @DrissiReda?

**Comment by DrissiReda** [comment]
That sounds great. Is there a way to verify which settings are getting overridden? Besides the solution you proposed of logging a warning every time OS finds an overridden setting in the config file.

**Comment by dblock** [comment]
> That sounds great. Is there a way to verify which settings are getting overridden?

I don't know from the top of my head.

**Comment by dblock** [comment]
Let's reopen this as "Log that a setting is being overridden and will not be used from oensearch.yml".

**Comment by JalODan** [comment]
When sending the same request, why would this setting be not recognized?
![image](https://user-images.githubusercontent.com/58619914/230704848-51c1ba6f-7d74-4b81-a468-355cab417fe2.png)

**Comment by sandeshkr419** [comment]
[Search Triage] We do not plan to incorporate this as of now, as we do not support compatibility with current ES versions. Closing this for now. If anyone as any future plans to incorporate something like this, please feel free to open up a feature request for the same.
